components/dashboard/src/admin/TeamDetail.tsx

```diff
@@ -15,24 +15,63 @@ import Label from "./Label";
 import Property from "./Property";
 import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
 import { BillingMode } from "@gitpod/gitpod-protocol/lib/billing-mode";
+import {
+    CostCenterJSON,
+    CostCenter_BillingStrategy,
+    ListUsageResponse,
+    Ordering,
+} from "@gitpod/gitpod-protocol/lib/usage";
+import Modal from "../components/Modal";
 
 export default function TeamDetail(props: { team: Team }) {
     const { team } = props;
     const [teamMembers, setTeamMembers] = useState<TeamMemberInfo[] | undefined>(undefined);
     const [billingMode, setBillingMode] = useState<BillingMode | undefined>(undefined);
     const [searchText, setSearchText] = useState<string>("");
+    const [costCenter, setCostCenter] = useState<CostCenterJSON>();
+    const [usageBalance, setUsageBalance] = useState<number>(0);
+    const [usageLimit, setUsageLimit] = useState<number>();
+    const [editSpendingLimit, setEditSpendingLimit] = useState<boolean>(false);
+    const [creditNote, setCreditNote] = useState<{ credits: number; note?: string }>({ credits: 0 });
+    const [editAddCreditNote, setEditAddCreditNote] = useState<boolean>(false);
+    const [usage, setUsage] = useState<ListUsageResponse>();
 
-    useEffect(() => {
+    const initialize = () => {
         (async () => {
             const members = await getGitpodService().server.adminGetTeamMembers(team.id);
             if (members.length > 0) {
                 setTeamMembers(members);
             }
         })();
         getGitpodService()
-            .server.adminGetBillingMode(AttributionId.render({ kind: "team", teamId: props.team.id }))
+            .server.adminGetBillingMode(AttributionId.render({ kind: "team", teamId: team.id }))
             .then((bm) => setBillingMode(bm));
-    }, [team]);
+        const attributionId = AttributionId.render(AttributionId.create(team));
+        getGitpodService().server.adminGetBillingMode(attributionId).then(setBillingMode);
+        getGitpodService().server.adminGetCostCenter(attributionId).then(setCostCenter);
+        getGitpodService().server.adminGetUsageBalance(attributionId).then(setUsageBalance);
+    };
+
+    useEffect(initialize, [team]);
+
+    useEffect(() => {
+        setUsageLimit(costCenter?.spendingLimit);
+        const attributionId = AttributionId.render(AttributionId.create(team));
+        getGitpodService()
```

Comment:
Q: should that really run no `costCenter` equals `undefined`?
resetting to some default values might make sense, but calling server seems unintended.

nit: `team` in dependency list is not relevant as its a non-optional prop.